### ISSUE TITLE
IOwinContext.Response.Write() ignored from CommandResultExtensions.Apply() method

### ISSUE THREAD

**Issue by akiander** [KentorIT/authservices#321]
I am attempting to use the **Kentor.AuthServices.Owin** library to authenticate with a SAML Identity Provider. We plug it into the latest version of **IdentityServer3** as one of two possible Identity Providers. 

The plug-in works perfectly when I run it locally on my development machine launched from Visual Studio using IIS Express. 

When I deploy the application to our Windows Server 2012 R2 web server (which uses IIS), then when I attempt to log in, the process never redirects to the Identity Provider. 

At the point in which it should write out the form that posts itself to the Identity Provider, nothing is written out the response stream so the browser just sits there with a blank page. 

I pulled down the latest Kentor.AuthServices code and put diagnostics information throughout the entire process and found that [this line of code](https://github.com/KentorIT/authservices/blob/master/Kentor.AuthServices.Owin/CommandResultExtensions.cs#L35) is where we should get the form written out, but the response stream never gets the form. 

So when I debug it, the markup for the form is there and appears to be fully populated and valid at the point in time when it is written to the Response, but it never gets written to the actual response ... so it never ends up in the browser and no redirect ever happens. 

I can see no errors generated anywhere, it just simply fails silently by not writing this form out, and the process stops with an empty browser window. 

Can someone help me to understand what might be failing here? What might cause the **IOwinContext.Response.Write()** line of code to get ignored, and what might I do to diagnose it or solve it? I'm open to any and all thoughts on what could be going wrong.

**Comment by albinsunnanbo** [comment]
It looks like it could be related to #295

In that case you could help us verify the fix in the branch https://github.com/KentorIT/authservices/compare/295ContentLength

**Comment by akiander** [comment]
Thanks for the quick reply. I took the fix from that branch and applied it to the local copy of Kentor.AuthServices.Owin that I have, but sadly, it did not fix this issue. 

I'm running on a very recent (2 day old I believe) copy of Kentor.AuthServices because I had to add logging internally to discover what was failing. When I applied the code above that nulls out the Content Length, it still does not send any response content. 

I might try to pull down the full version and try again before declaring defeat, but initial testing does not look promising. 

Let me know if you have any other thoughts on this, I'm open and available for suggestions here.
